src/js/components/DateInput/DateInput.js

```diff
@@ -0,0 +1,183 @@
+            placeholder: part,
+            length: [1, part.length],
+            regexp: new RegExp(`^[0-9]{1,${part.length}}$`),
+          };
+        }
+        return { fixed: part };
+      });
+    }, [schema]);
+
+    // textValue is only used when a format is provided
+    const [textValue, setTextValue] = useState(
+      schema ? valueToText(value, schema) : undefined,
+    );
+
+    // when format and not inline, wether to show the Calendar in a Drop
+    const [open, setOpen] = useState();
+
+    const range = Array.isArray(value);
+
+    const calendar = (
+      <Calendar
+        ref={inline ? ref : undefined}
+        id={inline && !format ? id : undefined}
+        range={range}
+        date={range ? undefined : value}
+        dates={range ? [value] : undefined}
+        onSelect={
+          disabled
+            ? undefined
+            : nextValue => {
+                let normalizedValue;
+                if (range && Array.isArray(nextValue))
+                  [normalizedValue] = nextValue;
+                // clicking an edge date removes it
+                else if (range) normalizedValue = [nextValue, nextValue];
+                else normalizedValue = nextValue;
+                if (schema) setTextValue(valueToText(normalizedValue, schema));
+                setValue(normalizedValue);
+                if (onChange) onChange({ value: normalizedValue });
+                if (open && !range) setOpen(false);
+              }
+        }
+        {...calendarProps}
+      />
+    );
+
+    if (!format) {
+      // When no format is specified, we don't give the user a way to type
+      if (inline) return calendar;
+
+      return (
+        <DropButton
+          ref={ref}
+          dropProps={{ align: { top: 'bottom', left: 'left' }, ...dropProps }}
+          dropContent={calendar}
+          icon={<CalendarIcon />}
+          {...buttonProps}
+        />
+      );
+    }
+
+    const input = (
+      <FormContext.Provider
+        // don't let MaskedInput drive the Form
+        value={{ useFormInput: (_, val) => [val, () => {}] }}
+      >
+        <MaskedInput
+          ref={ref}
+          name={name}
+          icon={<CalendarIcon />}
+          reverse
+          disabled={disabled}
+          mask={mask}
+          {...inputProps}
+          {...rest}
+          value={textValue}
+          onChange={event => {
+            const nextTextValue = event.target.value;
+            setTextValue(nextTextValue);
+            const nextValue = textToValue(nextTextValue, schema);
+            if (nextValue) {
+              // valid value
+              setValue(nextValue);
+              if (onChange) {
+                event.persist(); // extract from React synthetic event pool
+                const adjustedEvent = event;
+                adjustedEvent.value = nextValue;
+                onChange(adjustedEvent);
+              }
+            }
+          }}
+          onFocus={() => setOpen(true)}
```

Comment:
Should we also make a call to the users `onFocus()`?

Reply:
yes, added